addons/website_sale/models/product_product.py

```diff
@@ -215,68 +215,19 @@ def _get_extra_image_1920_urls(self):
             if extra_image.image_1920  # only images, no video urls
         ]
 
-    def _prepare_gmc_items(self):
-        """ Prepare Google Merchant Center items' fields.
-
-        See Google's (https://support.google.com/merchants/answer/7052112) documentation for more
-        information about each field.
-
-        Note: Depends on:
-            - `request.pricelist` to compute price and shipping informations,
-            - `request.website` to compute links, and
-            - `request.lang` to compute text based information (name, description, etc.) and links
-
-        :return: a dictionary for each product in this recordset.
-        :rtype: list[dict]
-        """
-        self = self.with_context(lang=request.lang.code)
-        base_url = request.website.get_base_url()
-
-        def format_product_link(url_):
-            url_ = urlparse(url_)._replace(query=f'pricelist={request.pricelist.id}').geturl()
-            return urljoin(base_url, self.env['ir.http']._url_lang(url_))
-
-        delivery_methods_sudo = self.env['delivery.carrier'].sudo().search(
+    @staticmethod
+    def _get_published_delivery_methods():
+        return request.env['delivery.carrier'].sudo().search(
             [('is_published', '=', True), ('website_id', 'in', (request.website.id, False))],
         )
-        all_countries = self.env['res.country'].search([])
-
-        return {
-            product: {
-                'id': product.default_code or product.id,
-                'title': product.with_context(display_default_code=False).display_name,
-                'description': product.website_meta_description or product.description_sale,
-                'link': format_product_link(product.website_url),
-                **product._prepare_gmc_identifier(),
-                **product._prepare_gmc_image_links(base_url),
-                **price_info,
-                **product._prepare_gmc_shipping_info(delivery_methods_sudo, all_countries),
-                **product._prepare_gmc_stock_info(),
-                **product._prepare_gmc_additional_info(),
-            }
-            for product in self
-            if product._is_variant_possible() and (price_info := product._prepare_gmc_price_info())
-        }
 
-    def _prepare_gmc_identifier(self):
-        """ Prepare the product identifiers for Google Merchant Center.
-
-        :return: The barcode of the product as GTIN
-        :rtype: dict
-        """
-        self.ensure_one()
-        if self.barcode:
-            return {'gtin': self.barcode, 'identifier_exists': 'yes'}
-        return {'identifier_exists': 'no'}
-
-    def _prepare_gmc_image_links(self, base_url):
-        """ Prepare the product image links for Google Merchant Center.
-
-        :return: The main product image link, and the extra images. No videos.
-        :rtype: dict
-        """
+    def _prepare_common_feed_items(self, format_product_link, base_url):
```

Comment:
This method is named _prepare..._items but only handles one product? What do you think about 

```suggestion
    def _prepare_feed_item_common(self, format_product_link, base_url):
```